src/App.tsx

```diff
@@ -7,33 +8,85 @@
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type ClockProps = {
+  name: string;
+};
+
+type ClockState = {
+  time: string;
+  hasClock: boolean;
+  name: string
+};
+
+export class App extends React.Component<ClockProps, ClockState> {
+  timerId: number | null = null;
+
+  state = {
+    time: '',
+    hasClock: true,
+    name: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.hideClock);
+    document.addEventListener('click', this.showClock);
+
+    this.updateTime();
+    this.timerId = window.setInterval(this.updateTime, 1000);
+    window.setInterval(this.updateClockName, 3300);
```

Comment:
The interval for updating the clock name should be cleared in componentWillUnmount.